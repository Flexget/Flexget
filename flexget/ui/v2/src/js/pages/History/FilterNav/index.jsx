import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IconButton from 'material-ui/IconButton';
import Select from 'material-ui/Select';
import Input, { InputLabel } from 'material-ui/Input';
import { MenuItem } from 'material-ui/Menu';
import SecondaryNav from 'common/SecondaryNav';
import { Spacer } from 'common/styles';
import { RotatingIcon, PaddedFormControl } from './styles';

const ENTER_KEY = 13;
const groupByFields = [
  {
    value: 'task',
    label: 'Task',
  },
  {
    value: 'time',
    label: 'Time',
  },
];

const sortByFields = [
  {
    value: 'details',
    label: 'Details',
  },
  {
    value: 'filename',
    label: 'Filename',
  },
  {
    value: 'id',
    label: 'ID',
  },
  {
    value: 'task',
    label: 'Task',
  },
  {
    value: 'time',
    label: 'Time',
  },
  {
    value: 'title',
    label: 'Title',
  },
  {
    value: 'url',
    label: 'URL',
  },
];

class FilterNav extends Component {
  static propTypes = {
    order: PropTypes.oneOf(['asc', 'desc']).isRequired,
    grouping: PropTypes.oneOf(groupByFields.map(({ value }) => value)).isRequired,
    sort: PropTypes.oneOf(sortByFields.map(({ value }) => value)).isRequired,
    toggleOrder: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
  };

  state = {
    task: '',
  }

  handleChange = (event) => {
    this.setState({ task: event.target.value });
  }

  handleKeyPress = (event) => {
    if (event.which === ENTER_KEY) {
      this.props.handleChange('task')(event);
    }
  }

  render() {
    const {
      order,
      grouping,
      sort,
      toggleOrder,
      handleChange,
    } = this.props;

    const {
      task,
    } = this.state;

    return (
      <SecondaryNav>
        <PaddedFormControl>
          <InputLabel htmlFor="task-filter">
            Filter By Task
          </InputLabel>
          <Input
            id="task-filter"
            placeholder="Task Name"
            value={task}
            onChange={this.handleChange}
            onKeyPress={this.handleKeyPress}
          />
        </PaddedFormControl>
        <Spacer />
        <PaddedFormControl>
          <InputLabel htmlFor="sort-by">Sort By</InputLabel>
          <Select
            value={sort}
            onChange={handleChange('sort')}
            input={<Input id="sort-by" />}
          >
            {sortByFields.map(({ value, label }) => (
              <MenuItem value={value} key={value}>{label}</MenuItem>
            ))}
          </Select>
        </PaddedFormControl>
        <PaddedFormControl>
          <InputLabel htmlFor="group-by">Group By</InputLabel>
          <Select
            value={grouping}
            onChange={handleChange('grouping')}
            input={<Input id="group-by" />}
          >
            {groupByFields.map(({ value, label }) => (
              <MenuItem value={value} key={value}>{label}</MenuItem>
            ))}
          </Select>
        </PaddedFormControl>
        <IconButton color="inherit" onClick={toggleOrder}>
          <RotatingIcon rotate={order === 'desc'} className="fa fa-chevron-up" />
        </IconButton>
      </SecondaryNav>
    );
  }
}

export default FilterNav;
