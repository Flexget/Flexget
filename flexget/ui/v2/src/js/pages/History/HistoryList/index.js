import { connect } from 'react-redux';
import { getGroupedHistory } from 'store/history/selectors';
import HistoryList from './HistoryList';

export function mapStateToProps({ history }, { grouping }) {
  return {
    history: getGroupedHistory(history, grouping),
    hasMore: !history.totalCount || history.items.length < history.totalCount,
  };
}

export default connect(mapStateToProps)(HistoryList);
export { HistoryList };
