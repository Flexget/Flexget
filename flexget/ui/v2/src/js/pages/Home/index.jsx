import React from 'react';
import InfoCard from './InfoCard';

const Home = () => <InfoCard />;

export default Home;
