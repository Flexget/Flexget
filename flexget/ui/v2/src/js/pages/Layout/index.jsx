import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'pages/Layout/Logo';
import Navbar from 'pages/Layout/Navbar';
import SideNav from 'pages/Layout/SideNav';
import LoadingBar from 'common/LoadingBar';
import ErrorStatus from 'common/ErrorStatus';
import InfoStatus from 'common/InfoStatus';
import {
  Wrapper,
  Main,
  Header,
  LogoWrapper,
  Nav,
  SideBar,
  Content,
} from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    loggedIn: false,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState({
      sideBarOpen: !this.state.sideBarOpen,
    });
  }

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav
              sideBarOpen={sideBarOpen}
              toggle={this.toggleSideBar}
            />
          </SideBar>
          <Content open={sideBarOpen}>
            { children }
          </Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
