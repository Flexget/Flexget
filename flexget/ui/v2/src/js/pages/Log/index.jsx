import React from 'react';
import { PaperWrapper } from 'common/styles';
import Header from './Header';
import LogTable from './LogTable';
import { TableWrapper } from './styles';

const LogPage = () => (
  <PaperWrapper elevation={4}>
    <Header />
    <TableWrapper>
      <LogTable />
    </TableWrapper>
  </PaperWrapper>
);

export default LogPage;
