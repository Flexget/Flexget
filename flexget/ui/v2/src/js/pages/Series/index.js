import { connect } from 'react-redux';
import {
  GET_SHOWS,
} from 'store/series/shows/actions';
import { request } from 'utils/actions';
import Series from './Series';

export function mapStateToProps({ series }) {
  return {
    shows: series.shows.items,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getShows: payload => dispatch(request(GET_SHOWS, payload)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Series);
export { Series };
