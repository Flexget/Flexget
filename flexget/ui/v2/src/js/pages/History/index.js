import { connect } from 'react-redux';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'store/history/actions';
import History from './History';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(History);
export { History };
