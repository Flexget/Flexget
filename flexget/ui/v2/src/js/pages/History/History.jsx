import React, { Component } from 'react';
import { findDOMNode } from 'react-dom';
import PropTypes from 'prop-types';
import HistoryList from 'pages/History/HistoryList';
import FilterNav from 'pages/History/FilterNav';
import { NoPaddingWrapper } from 'common/styles';

export default class HistoryPage extends Component {
  static propTypes = {
    getHistory: PropTypes.func.isRequired,
  };

  state = {
    grouping: 'time',
    sort: 'time',
    order: 'desc',
  };

  getHistory = page => this.props.getHistory({
    page,
    sort_by: this.state.sort,
    order: this.state.order,
    task: this.state.task,
  });

  setScroll = (node) => { this.scroll = node; }
  restartLoader = () => {
    findDOMNode(this.scroll).scrollIntoView(); // eslint-disable-line react/no-find-dom-node
    this.getHistory(1);
    this.scroll.pageLoaded = 1;
  }

  handleChange = key => event => this.setState({ [key]: event.target.value }, this.restartLoader)
  toggleOrder = () => this.setState({
    order: (this.state.order === 'asc' ? 'desc' : 'asc'),
  }, this.restartLoader)

  render() {
    const { grouping } = this.state;

    return (
      <NoPaddingWrapper>
        <FilterNav
          handleChange={this.handleChange}
          toggleOrder={this.toggleOrder}
          {...this.state}
        />
        <HistoryList
          grouping={grouping}
          getHistory={this.getHistory}
          setScroll={this.setScroll}
          ref={(node) => { this.list = node; }}
        />
      </NoPaddingWrapper>
    );
  }
}
