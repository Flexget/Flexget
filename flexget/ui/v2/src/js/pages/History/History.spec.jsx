import React from 'react';
import { shallow } from 'enzyme';
import { History } from 'pages/History';

describe('pages/History', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <History
        getHistory={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
