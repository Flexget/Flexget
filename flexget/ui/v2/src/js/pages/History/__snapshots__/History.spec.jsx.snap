// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/History renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HistoryPage
    getHistory={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FilterNav
          grouping="time"
          handleChange={[Function]}
          order="desc"
          sort="time"
          toggleOrder={[Function]}
        />,
        <Connect(HistoryList)
          getHistory={[Function]}
          grouping="time"
          setScroll={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "grouping": "time",
          "handleChange": [Function],
          "order": "desc",
          "sort": "time",
          "toggleOrder": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "getHistory": [Function],
          "grouping": "time",
          "setScroll": [Function],
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FilterNav
            grouping="time"
            handleChange={[Function]}
            order="desc"
            sort="time"
            toggleOrder={[Function]}
          />,
          <Connect(HistoryList)
            getHistory={[Function]}
            grouping="time"
            setScroll={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "grouping": "time",
            "handleChange": [Function],
            "order": "desc",
            "sort": "time",
            "toggleOrder": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "getHistory": [Function],
            "grouping": "time",
            "setScroll": [Function],
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
